### DISCUSSION TITLE
Adding a magnet region

### DISCUSSION THREAD

**Discussion by yyang262** [ansys/pymotorcad#374] [Q&A]
After creating a region, is there a way to set the region as a magnet region and set the magnetic properties such as br, polarity, etc?

**Comment by JackB-Ansys** [comment]
Hello, currently it is not recommended to create magnet regions from scratch. The best practice is to set up the standard Motor-CAD template with all the magnet regions that you need, and then modify the magnets with the adaptive templates script.
You can set the magnet attributes/properties (angles, Br etc.) in the Adaptive Template script using the appropriate methods: https://motorcad.docs.pyansys.com/version/stable/methods/_autosummary_geometry_methods/ansys.motorcad.core.geometry.RegionMagnet.html
